src/Kiota.Builder/Refiners/CSharpRefiner.cs

```diff
@@ -257,4 +258,19 @@ protected static void CorrectIndexerType(CodeIndexer currentIndexer)
                 })
         },
     };
+
+    private void SetTypeAccessModifiers(CodeElement currentElement)
+    {
+        switch (currentElement)
```

Comment:
You can simplify the implementation here if you make the code dom elements that have the access property implement IAccessibleElement

Reply:
Implemented.

Reply:
What do you think of this?
```suggestion
       if(currentElement is IAccessibleElement accessibleElement and (CodeEnum or CodeClass))
            accessibleElement.Access = _configuration.TypeAccessModifier;
```